Partial fix for small buffer pool 100% cpu usage lp:1433432

ihanick opened the pull request:
https://bugs.launchpad.net/percona-server/+bug/1433432

http://jenkins.percona.com/job/percona-server-5.6-param/1034/

Using --innodb_empty_free_list_algorithm=legacy to avoid deadlock
Increase lru page cleaner sleep time if free lists filled less than 1% and previous clean batch had nothing to clean

laurynas-biveinis commented:
It needs a descriptive commit message. Specifically, that this does not resolve deadlock as such but only needless CPU spinning in the LRU manager thread when it's unable to flush

laurynas-biveinis commented on storage/innobase/buf/buf0flu.cc:
```diff
@@ -2608,17 +2610,20 @@ static
-	if (free_len < max_free_len / 100) {
+	if (free_len < max_free_len / 100 && lru_n_flushed) {
 
 		/* Free lists filled less than 1%, no sleep */
```
" ... and last iteration was able to flush " or similar update for the comment

laurynas-biveinis commented on storage/innobase/buf/buf0flu.cc:
```diff
@@ -2608,17 +2610,20 @@ static
 		*lru_sleep_time = 0;
-	} else if (free_len > max_free_len / 5) {
+	} else if (free_len > max_free_len / 5
+		   ||(free_len < max_free_len / 100 && lru_n_flushed == 0)) {
```
Formatting ||< space>(

laurynas-biveinis commented on storage/innobase/buf/buf0flu.cc:
```diff
@@ -2608,17 +2610,20 @@ static
+	} else if (free_len > max_free_len / 5
+		   ||(free_len < max_free_len / 100 && lru_n_flushed == 0)) {
 
 		/* Free lists filled more than 20%, sleep a bit more */
```
This comment also needs an update

ihanick commented:
- Added fix description to commit message
- fixed || formatting
- updated added comments for each modified if branches in page_cleaner_adapt_lru_sleep_time
